src/iuse_actor.cpp

```diff
@@ -3104,3 +3105,41 @@ long place_trap_actor::use( player * const p, item * const it, bool, const tripo
     }
     return 1;
 }
+
+void emit_actor::load( JsonObject &obj )
+{
+    assign( obj, "emits", emits );
+    assign( obj, "scale_qty", scale_qty );
+}
+
+long emit_actor::use( player*, item *it, bool, const tripoint &pos ) const
+{
+    const float scaling = scale_qty ? it->charges : 1;
```

Comment:
A better solution would be to make `scale_qty` a boolean (default `true`) and then run the loop below once per unit volume (`250ml`). It achieves everything your aiming for without either affecting the standard nature of `emit` sources or requiring specific configuration in the most common cases